internal/common/common.go

```diff
@@ -41,50 +42,77 @@ func init() {
 	}
 }
 
-func GetVaultAuthAndTarget(ctx context.Context, c client.Client, obj client.Object) (*secretsv1beta1.VaultAuth, types.NamespacedName, error) {
-	var authRef string
+func GetAuthAndTargetNamespacedName(obj client.Object) (types.NamespacedName, types.NamespacedName, error) {
+	var authRef types.NamespacedName
 	var target types.NamespacedName
-	switch o := obj.(type) {
-	case *secretsv1beta1.VaultPKISecret:
-		authRef = o.Spec.VaultAuthRef
-		target = types.NamespacedName{
-			Namespace: o.Namespace,
-			Name:      o.Name,
-		}
-	case *secretsv1beta1.VaultStaticSecret:
-		authRef = o.Spec.VaultAuthRef
-		target = types.NamespacedName{
-			Namespace: o.Namespace,
-			Name:      o.Name,
-		}
-	case *secretsv1beta1.VaultDynamicSecret:
-		authRef = o.Spec.VaultAuthRef
-		target = types.NamespacedName{
-			Namespace: o.Namespace,
-			Name:      o.Name,
-		}
-	default:
-		return nil, types.NamespacedName{}, fmt.Errorf("unsupported type %T", o)
+
+	m, err := NewSyncableSecretMetaData(obj)
+	if err != nil {
+		return types.NamespacedName{}, types.NamespacedName{}, fmt.Errorf("unsupported type %T", obj)
 	}
 
-	var authName types.NamespacedName
-	if authRef == "" {
+	if m.AuthRef != "" {
+		// todo: some sorta sanity checking?
+		names := strings.Split(m.AuthRef, "/")
+		if len(names) > 1 {
```

Comment:
Probably want to return an error if `len(names) > 2`

Reply:
Fixed in a52da2d